x/params/store/store.go

```diff
@@ -0,0 +1,176 @@
+package store
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Individual parameter store for each keeper
+// Transient store persists for a block, so we use it for
+// recording whether the parameter has been changed or not
+type Store struct {
+	cdc  *codec.Codec
+	key  sdk.StoreKey // []byte -> []byte, stores parameter
+	tkey sdk.StoreKey // []byte -> bool, stores parameter change
+
+	space []byte
+}
+
+// NewStore constructs a store with namestore
+func NewStore(cdc *codec.Codec, key sdk.StoreKey, tkey sdk.StoreKey, space string) Store {
+	return Store{
+		cdc:  cdc,
+		key:  key,
+		tkey: tkey,
+
+		space: []byte(space),
```

Comment:
Lets actually provide a larger capacity here. This will allow prefixing to be more efficient, within the append.